anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java

```diff
@@ -0,0 +1,108 @@
+package org.stellar.anchor.reference.event;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.AmazonSQSAsyncClientBuilder;
+import com.amazonaws.services.sqs.model.*;
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.concurrent.*;
+import javax.annotation.PreDestroy;
+import org.stellar.anchor.event.models.QuoteEvent;
+import org.stellar.anchor.event.models.TransactionEvent;
+import org.stellar.anchor.reference.config.SqsListenerSettings;
+import org.stellar.anchor.util.Log;
+
+public class SqsListener extends AbstractEventListener {
+  private final AnchorEventProcessor processor;
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(5);
+  private final AmazonSQSAsync sqsClient;
+
+  public SqsListener(SqsListenerSettings sqsListenerSettings, AnchorEventProcessor processor) {
+    this.processor = processor;
+    this.sqsClient =
+        AmazonSQSAsyncClientBuilder.standard()
+            .withRegion(sqsListenerSettings.getRegion())
+            .withCredentials(
+                new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(
+                        sqsListenerSettings.getAccessKey(), sqsListenerSettings.getSecretKey())))
+            .build();
+    SqsListenerSettings.Queues q = sqsListenerSettings.getEventTypeToQueue();
+    if (sqsListenerSettings.isUseSingleQueue()) {
+      executor.scheduleAtFixedRate(new SqsListen(q.getAll()), 0, 10, TimeUnit.SECONDS);
+    } else {
+      executor.scheduleAtFixedRate(
+          new SqsListen(q.getTransactionPaymentReceived()), 0, 10, TimeUnit.SECONDS);
+      executor.scheduleAtFixedRate(new SqsListen(q.getQuoteCreated()), 0, 10, TimeUnit.SECONDS);
+      executor.scheduleAtFixedRate(
+          new SqsListen(q.getTransactionCreated()), 0, 10, TimeUnit.SECONDS);
+      executor.scheduleAtFixedRate(new SqsListen(q.getTransactionError()), 0, 10, TimeUnit.SECONDS);
+    }
+  }
+
+  public List<Message> getSqsMessages(String queueUrl) {
+    ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest();
+    receiveMessageRequest.setQueueUrl(queueUrl);
+    receiveMessageRequest.setMaxNumberOfMessages(
+        10); // SQS supports a max of 10 messages per request
+    receiveMessageRequest.withMessageAttributeNames("anchor-event-class");
+    ReceiveMessageResult receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
+    return receiveMessageResult.getMessages();
+  }
+
+  class SqsListen implements Runnable {
+    private final String queue;
+
+    public SqsListen(String queue) {
+      this.queue = queue;
+    }
+
+    public void run() {
+      Log.debug("SQS event consumer started for queue: " + queue);
```

Comment:
`Log.info`?

We will need to define the logging strategy for `debug`, `info`, `warning` and `error`.